iothub/device/src/Transport/DeviceIdentity.cs

```diff
@@ -38,6 +42,25 @@ internal DeviceIdentity(IotHubConnectionString iotHubConnectionString, AmqpTrans
                 AuthenticationModel = AuthenticationModel.X509;
             }
         }
+        private static string CreateAudience(IotHubConnectionString connectionString)
+        {
+            if (connectionString.SharedAccessKeyName.IsNullOrWhiteSpace())
+            {
+                if (connectionString.ModuleId.IsNullOrWhiteSpace())
+                {
+                    return $"{connectionString.HostName}/devices/{WebUtility.UrlEncode(connectionString.DeviceId)}";
```

Comment:
Please pick-up from `master`: this appears to be missing recent fixes.

Reply:
done

Reply:
done